eng/pipelines/merge-docs-main-to-live.yml

```diff
@@ -0,0 +1,92 @@
+parameters:
+- name: WhatIfPreference
+  type: boolean
+  default: false
+
+- name: DocsRepos
+  type: object
+  default:
+    - RepoName: azure-docs-sdk-java
+      RepoOwner: Azure
+      AllowlistPath: docsms-allowlist/java-allowlist.txt
+    - RepoName: azure-docs-sdk-python
+      RepoOwner: MicrosoftDocs
+      AllowlistPath: docsms-allowlist/python-allowlist.txt
+    - RepoName: azure-docs-sdk-node
+      RepoOwner: MicrosoftDocs
+      AllowlistPath: docsms-allowlist/js-allowlist.txt
+
+- name: TargetBranch
+  type: string
+  default: live
+
+jobs:
+- ${{ each repo in parameters.DocsRepos }}:
+  - job: 
+    displayName: 'ReleaseDocsMs: ${{repo.RepoName}}'
+    pool:
+      name: azsdk-pool-mms-ubuntu-2004-general
+      vmImage: MMSUbuntu20.04
+    variables:
+      DocRepoLocation: $(Pipeline.Workspace)/docs
+    steps:
+      # Checkout tool repo
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+      # Checkout docs repo
+      - template: /eng/common/pipelines/templates/steps/sparse-checkout.yml
+        parameters:
+          SkipDefaultCheckout: true
+          Repositories:
+          - Name: ${{ repo.RepoOwner }}/${{ repo.RepoName }}
+            WorkingDirectory: $(DocRepoLocation)
+          Paths:
+          - /*
+
+      - task: Powershell@2
+        inputs:
+          pwsh: true
+          filePath: eng/scripts/create-tag-for-current-branch.ps1
+          arguments: 
+            -ReleaseTag "live_release-$(Get-Date -Format 'yyyy-MM-dd').$(Build.BuildId)"
+            -GithubUrl "https://$(azuresdk-github-pat)@github.com/${{repo.RepoOwner}}/${{repo.RepoName}}.git" `
+            -WhatIf:$${{ parameters.WhatIfPreference }}
+          workingDirectory: $(DocRepoLocation)
+        displayName: Create tag for base branch
+
+      - template: /eng/common/pipelines/templates/steps/set-default-branch.yml
+        parameters:
+          WorkingDirectory: $(DocRepoLocation)
+          DefaultBranchVariableName: ${{repo.RepoName}}
+
+      - task: Powershell@2
+        inputs:
+          pwsh: true
+          filePath: eng/scripts/validate-changes-on-allowed-path.ps1
+          arguments: 
+            -TargetBranch ${{ parameters.TargetBranch }} `
+            -BaseBranch '$(${{repo.RepoName}})' `
+            -AllowListPath "${{repo.AllowlistPath}}"
+          workingDirectory: $(DocRepoLocation)
+        displayName: Validate the changes within allowlist
+        condition: ne(variables['Skip.Validation'], 'true')
+
+      - task: Powershell@2
+        inputs:
+          pwsh: true
+          filePath: eng/scripts/sync-base-branch-to-target.ps1
+          arguments: 
+            -TargetBranch ${{ parameters.TargetBranch }} `
+            -BaseBranch "$(${{repo.RepoName}})" 
+          workingDirectory: $(DocRepoLocation)
+        displayName: Sync base branch to target
+
+      - template: /eng/common/pipelines/templates/steps/git-push-changes.yml
```

Comment:
Don't you need to set the HasChanges variable somewhere as a workaround otherwise this will not do anything.

Reply:
I thought the set variable step looks redundant in code and if we want to get rid of the change maybe delete in pipeline UI is the easiest way. But it is hard for the cleanup work. I added a step for the variable setting.